xbmc/guilib/Texture.h

```diff
@@ -91,7 +91,7 @@ class CBaseTexture
 
   virtual void CreateTextureObject() = 0;
   virtual void DestroyTextureObject() = 0;
-  virtual void LoadToGPU() = 0;
+  virtual void LoadToGPU(bool bFreeSysMem = true) = 0;
```

Comment:
I don't like this default parameter hacks. From a design POV those have zero value. You only do this to avoid touching more code. Whenever I hit one of those hacks, I remove the default.

Why don't you add a new method for your purpose and factor out the common code into a separate protected method. Don't bother the rest of the app with a paramter is does not use.

Reply:
Updated. The new commit adds a new method instead of using a default parameter in `LoadToGPU()`. How's that look?